app/main.py

```diff
@@ -1,2 +1,54 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 2 / 3) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index: int = self._hash(key) % self.capacity
+        if not self.table[index]:
+            self.table[index] = [(key, value)]
+            self.size += 1
+        else:
+            pairs = self.table[index]
```

Comment:
it's better to make if self.table[index] else (case when not), so that it's more readable